packages/model-viewer/src/test/features/scene-graph-spec.ts

```diff
@@ -85,6 +89,41 @@ suite('ModelViewerElementBase with SceneGraphMixin', () => {
         expect(glTFroot.children[1].userData.variantMaterials.size).to.be.eq(3);
       });
 
+      test(
+          `Setting varianName to null results in primitive
+           reverting to default/initial material`,
+          async () => {
+            let defaultIndex = 0;
+            // Patches the first primitive with variant-info and a default
+            // material index of 0.
+            for (const primitive of element.model![$primitives]) {
+              if (primitive.variantInfo != null &&
+                  primitive[$defaultMaterialIdx] == 0) {
+                // Override enable variant to watch for change.
+                const override: PrimitiveNode&{superSetActiveMaterial: any} =
+                    primitive as PrimitiveNode & {superSetActiveMaterial: any};
+                override.superSetActiveMaterial = primitive.setActiveMaterial;
+
+                override.setActiveMaterial = async(material: number):
+                    Promise<ThreeMaterial|ThreeMaterial[]|null> => {
+                      defaultIndex = material;
+                      return override.superSetActiveMaterial(material);
```

Comment:
This seems pretty complicated. Might there be a way to just check the `element.model![primitive][0]` for its material, then check that it's different when you switch variants, then check it's back to the original value when you set `null`?

Reply:
Yeah that's what I mean when I say it's a bit tricky, we can access the primitives as you show but the order is not guaranteed at the moment so I can't rely on that, tests get flaky. However, I simplified things quite a bit, in this next update.